include/net/http_app.h

```diff
@@ -0,0 +1,1039 @@
+				      u8_t *personalization_data,
+				      size_t personalization_data_len,
+				      net_app_cert_cb_t cert_cb,
+				      net_app_entropy_src_cb_t entropy_src_cb,
+				      struct k_mem_pool *pool,
+				      k_thread_stack_t *stack,
+				      size_t stack_len)
+{
+	ARG_UNUSED(ctx);
+	ARG_UNUSED(server_banner);
+	ARG_UNUSED(personalization_data);
+	ARG_UNUSED(personalization_data_len);
+	ARG_UNUSED(cert_cb);
+	ARG_UNUSED(entropy_src_cb);
+	ARG_UNUSED(pool);
+	ARG_UNUSED(stack);
+	ARG_UNUSED(stack_len);
+
+	return -ENOTSUP;
+}
+#endif /* CONFIG_HTTPS */
+
+static inline int http_server_enable(struct http_ctx *ctx)
+{
+	ARG_UNUSED(ctx);
+	return -ENOTSUP;
+}
+
+static inline int http_server_disable(struct http_ctx *ctx)
+{
+	ARG_UNUSED(ctx);
+	return -ENOTSUP;
+}
+
+static inline
+struct http_root_url *http_server_add_url(struct http_server_urls *urls,
+					  const char *url, u8_t flags)
+{
+	ARG_UNUSED(urls);
+	ARG_UNUSED(url);
+	ARG_UNUSED(flags);
+
+	return NULL;
+}
+#endif /* CONFIG_HTTP_SERVER */
+
+#if defined(CONFIG_HTTP_CLIENT)
+/**
+ * @brief Generic function to send a HTTP request to the network. Normally
+ * applications would not need to use this function.
+ *
+ * @param ctx HTTP context.
+ * @param req HTTP request to perform.
+ * @param timeout Timeout when doing net_buf allocations.
+ * @param user_data User data related to this request. This is passed
+ * to send callback if user has specified that.
+ *
+ * @return Return 0 if ok, and <0 if error.
+ */
+int http_request(struct http_ctx *ctx,
+		 struct http_request *req,
+		 s32_t timeout,
+		 void *user_data);
+
+/**
+ * @brief Cancel a pending request.
+ *
+ * @param ctx HTTP context.
+ *
+ * @return Return 0 if ok, and <0 if error.
+ */
+int http_request_cancel(struct http_ctx *ctx);
+
+/**
+ * @brief Send a HTTP request to peer.
+ *
+ * @param http_ctx HTTP context.
+ * @param req HTTP request to perform.
+ * @param cb Callback to call when the response has been received from peer.
+ * @param response_buf Caller-supplied buffer where the HTTP response will be
+ * stored
+ * @param response_buf_len Length of the caller-supplied buffer.
+ * @param user_data A valid pointer on some user data or NULL
+ * @param timeout Amount of time to wait for a reply. If the timeout is 0,
+ * then we return immediately and the callback (if set) will be called later.
+ *
+ * @return Return 0 if ok, and <0 if error.
+ */
+int http_client_send_req(struct http_ctx *http_ctx,
+			 struct http_request *req,
+			 http_response_cb_t cb,
+			 u8_t *response_buf,
+			 size_t response_buf_len,
+			 void *user_data,
+			 s32_t timeout);
+
+/**
+ * @brief Send a HTTP GET request to peer.
+ *
+ * @param http_ctx HTTP context.
+ * @param url URL to use.
+ * @param host Host field in HTTP header. If set to NULL, then server
+ * name is used.
+ * @param extra_header_fields Any extra header fields that caller wants
+ * to add. This can be set to NULL. The format is "name: value\r\n"
```

Comment:
Can't use ``\r`` in a doxygen comment.  Recommend something like this instead:

     The format is "name: value<HTTP_CRLF>"

Reply:
Thanks David! Really peculiar issue indeed, especially as the warning did not match the root cause.

Reply:
> Can't use \r in a doxygen comment. 

Maybe it should be double-quoted, like `\\r`? @jukkar , please try that first, because having `\r\n` is clearer than `<HTTP_CRLF>`.